### PR TITLE
Migrate crud api codebase to typescript

### PR DESCRIPTION
**Pull request by justmic007**
- [X] Setup Typescript.
- [X] renamed all .js extension files in /src to .ts
- [X] replaced **babel-node** with **ts-node** for development.
- [X] Added typing to all private interface.

### CODE REVIEW

**Comment by Ehbraheem** [comment]
@noisyscanner Please take note of this new version of **crud-api**. Nothing much has been changed from public interface of **crud-api** except for the changes introduced by #45 and #43

**Comment by noisyscanner** [comment]
Thank you @Ehbraheem

File: lib/bootstrap.js [hunk 1]
```diff
@@ -1,113 +1,98 @@
...
-var _default = application;
-exports.default = _default;
\ No newline at end of file
+!module.parent &&
+    (async () => {
+        const app = await application();
+        await app.start();
+        console.log(`App runninng on ${app.info.uri}`); // eslint-disable-line
+    })();
+exports.default = application;
```
Comment (jabdul) [inline lib/bootstrap.js:98]: Tab space formatting seems off in this file.

File: src/users/services.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    config,
+  });
+
+  if (!result) {
+    // log(['users', 'error', 'database'], result);
+    throw Error(result);
+  }
+  // const {
+  //   firstnam, last
+  // }
```
Comment (jabdul) [inline src/users/services.ts:16]: could we remove these commented out code.

**Pull request merged**